src/Panel.jsx

```diff
@@ -23,22 +35,29 @@ var Panel = React.createClass({
     };
   },
 
-  handleSelect: function (e) {
-    if (this.props.onSelect) {
-      this._isChanging = true;
-      this.props.onSelect(this.props.eventKey);
-      this._isChanging = false;
+  handleSelect: function(e){
+    if(this.props.onSelect) {
+      this.props.onSelect(e, this.props.eventKey)
+    } else {
+      this.handleClickEvent(e);
+      this.handleToggle();
     }
```

Comment:
Following the discussion on #419 instead of an else call here we can deal with the toggle if `e.selected` is true, and leave the e.preventDefault decision to the dev handling the `onSelect` event. I'm thinking something like this:

```
handleSelect: function(e) {
  e.selected = true;

  if (this.props.onSelect) {
    this.props.onSelect(e, this.props.eventKey);
  }

  if (e.selected) {
    this.handleToggle();
  }
}
```